cache/bytes_limit_decorator.go

```diff
@@ -0,0 +1,66 @@
+package cache
+
+import (
+	"container/list"
+	"errors"
+
+	"github.com/cosmos/iavl/common"
+)
+
+type lruCacheWithBytesLimit struct {
+	lruCache
+	bytesLimit       int
+	curBytesEstimate int
```

Comment:
why is it called estimate? Should we add a comment that it may potentially slightly undercount, but we deem it fine?

Reply:
It is called estimate because I did not find a straightforward way to test actual allocations.

I tried grabbing memory from runtime to test but it was slightly off. I think it's because it doesn't produce the memory needed for allocating "slice metadata" - length, capacity and pointer to data.

This estimate is based on the knowledge of how slices and strings are represented in memory in Go. That's why I named it estimate.

I'll add a comment about this

Reply:
Oh gotcha. Yeah being off by a small constant for go memory layout details (which aren't guaranteed to be preserved across versions) is totally fine!

Reply:
The comment is added